src/player.cpp

```diff
@@ -5386,7 +5386,7 @@ std::vector<Creature *> player::get_targetable_creatures( const int range ) cons
         // TODO: get rid of fake npcs (pos() check)
         return this != &critter && pos() != critter.pos() && attitude_to( critter ) != Creature::Attitude::A_FRIENDLY &&
         round( rl_dist_exact( pos(), critter.pos() ) ) <= range &&
-        ( sees( critter ) || sees_with_infrared( critter ) );
+        ( ( sees( critter ) && g->m.sees( pos(), critter.pos(), 100 ) ) || sees_with_infrared( critter ) );
```

Comment:
There are so much conditions in here. Maybe split this one big check to several smaller checks?

Reply:
Good idea